dvc/ignore.py

```diff
@@ -172,14 +176,19 @@ def __init__(self, fs, root_dir):
         self.root_dir = root_dir
         self.ignores_trie_fs = Trie()
         self._ignores_trie_subrepos = Trie()
+        self.config = Config()
```

Comment:
This seems to be breaking the scope. Can't we just pass particular options when creating dvcignore in the repo?

Reply:
Btw, doing that will allow simplifying tests by not having to monkeypatch global config dir all the time.